### PR TITLE
Refactor modal for custom dimnames

### LINKED ISSUE #456: Recycle the heatmap UI for DotPlots to pick points to label.
We can recycle the heatmap feature selection modal to allow people to select the rows or columns they want to label in a dot plot via `geom_text`. This will allow people to mark features of interest in higher-order panels like MA plots.

### PR DESCRIPTION
**Pull request by kevinrue**
Fixes #456

### CODE REVIEW

**Comment by kevinrue** [comment]
- [ ] Also, the text UI should have a hr() on top of it
- [ ] Also, I wish we could get some more space between the edit button and the "Font size" label, it's a bit cramped.
- [ ] Also, if I type a random name and press "validate names", the name disappears rather than being validated. It is only after I click "apply" and then I go back to validate the name that the # appears.
- [ ] While you're there, you might as well explicitly mention it in the DotPlot tour, it's going to be too hidden for people to stumble on it.

File: R/family_DotPlot.R [hunk 1]
```diff
@@ -368,15 +371,28 @@ setMethod(".createObservers", "DotPlot", function(x, se, input, session, pObject
...
     tagList(
+        hr(),
+        checkboxInput(.input_FUN(.plotCustomLabels),
+                      label=sprintf("Label custom %ss", .singleSelectionDimension(x)),
+                      value=x[[.plotCustomLabels]]),
+        .conditional_on_check_solo(
+            .input_FUN(.plotCustomLabels),
+            on_select=TRUE,
+            actionButton(.input_FUN(.dimnamesModalOpen), label=sprintf("Edit %s names", .singleSelectionDimension(x)))),
+        br(),
```
Comment (LTLA) [inline R/family_DotPlot.R:395]: Move this `br()` into the `.conditional_on_check_solo`?

File: R/family_DotPlot.R [hunk 2]
```diff
@@ -594,6 +612,14 @@ setMethod(".generateDotPlot", "DotPlot", function(x, labels, envir) {
         violin_horizontal=do.call(.violin_plot, c(args, list(horizontal=TRUE))),
         scatter=do.call(.scatter_plot, args)
     )
+    
+    if (x[[.plotCustomLabels]]) {
+        N <- length(plot_cmds)
+        plot_cmds[[N]] <- paste(plot_cmds[[N]], "+")
+        dn <- .convert_text_to_names(x[[.plotCustomLabelsText]])
+        label_cmd <- sprintf('ggrepel::geom_text_repel(aes(x=X, y=Y, label=LabelBy), subset(plot.data, LabelBy %%in%% %s), min.segment.length = unit(0, "mm"))', .deparse_for_viewing(dn))
```
Comment (LTLA) [inline R/family_DotPlot.R]: Need a `grid::unit`, probably.

File: R/observers_modal.R [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+#' Observers are set up to monitor the UI elements that can change various parameters specific to the \code{ComplexHeatmapPlot} panel.
+#' A \code{NULL} is invisibly returned.
+#'
+#' @author Kevin Rue-Albrecht
+#' 
+#' @importFrom shiny modalDialog removeModal fluidRow column h4 actionButton br tagList em strong
+#' @importFrom shinyAce aceEditor updateAceEditor
+#' @importFrom shinyjs disabled
+#' @rdname INTERNAL_create_modal_observers_for_dimnames
+.create_modal_observers_for_dimnames <- function(plot_name, slot_name, button_name, se, input, session, pObjects, rObjects, source_type) {
```
Comment (LTLA) [inline R/observers_modal.R:25]: 4 spaces

File: R/family_DotPlot.R (hunk 1 above)
Comment (LTLA) [reply R/family_DotPlot.R:395]: Ah forget it, it doesn't work if it's inside the conditional.

File: R/observers_modal.R (hunk 3 above)
Comment (kevinrue) [reply R/observers_modal.R:25]: Done

**Pull request merged**